src/attrs.rs

```diff
@@ -86,16 +86,25 @@ impl quote::ToTokens for Attrs {
     }
 }
 
+const MERMAID_INIT_SCRIPT: &str = r#"
+    let docs_theme = currentTheme.href;
+    let amrn_dark = docs_theme.endsWith("dark.css") || docs_theme.endsWith("ayu.css") 
+    let amrn_theme = (amrn_dark ?? false) ? 'dark' : 'default';
```

Comment:
I believe you put `undefined` handling in the wrong place. The problem is that `currentTheme` could be not defined

Reply:
Yes, and then everything that used currentTheme will be undefined too. That's why I handle it where it matters for configuration of the `mermaid`. Creating some homebrew default variant of `currentTheme` doesn't make sense, especially since this being included someplace other then rustdoc isn't a supported case anyway.

The way I handle it here makes sure mermaid will receive correct input, no more, no less.

Reply:
Javascript doesn't have propagation of `undefined`. If `currentTheme` is not defined or set to `undefined`, the very first line will raise `TypeError`.

Reply:
Okay, that's fine with me -- if rustdoc changes API, the error will be obviously visible. Either way, it will select the theme correctly.

Reply:
Rewritten it in a more graceful manner, albeit it takes more space now.